### PR TITLE
Lowercase headers

### PR DESCRIPTION
**Pull request by arashnd**
## All PRs:

* [X] Has tests
* [x] Documentation updated

## Adding a new header

Generally, adding a new header is always OK.

* Is the header supported by any user agent? If so, which?
* What does it do?
* What are the valid values for the header?
* Where does the specification live?

## Adding a new CSP directive

* Is the directive supported by any user agent? If so, which?
* What does it do?
* What are the valid values for the directive?

---

Rack 3 `Response header keys can no longer include uppercase characters.`
https://github.com/rack/rack/blob/main/UPGRADE-GUIDE.md#rack-3-upgrade-guide

Rack Lint error out when using `secure_headers` because its adding headers with uppercase
```
Rack::Lint::LintError: uppercase character in header name: X-Frame-Options (Rack::Lint::LintError)
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,11 +62,11 @@ NOTE: this version is a breaking change due to the removal of HPKP. Remove the H
 
 ## 5.0.2
 
-- Updates `Referrer-Policy` header to support multiple policy values
```
Comment (rzhade3) [inline CHANGELOG.md:65]: We shouldn't retroactively changed the changelog here; it'll be difficult for folks to understand the timeline of changes. Instead, let's add a new release version that states that it lower cases all header keys.

File: README.md [hunk 2]
```diff
@@ -92,19 +92,19 @@ end
 ```
 
 ### Deprecated Configuration Values
-* `block_all_mixed_content` - this value is deprecated in favor of `upgrade_insecure_requests`. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content for more information.
+* `block_all_mixed_content` - this value is deprecated in favor of `upgrade_insecure_requests`. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/content-security-policy/block-all-mixed-content for more information.
```
Comment (rzhade3) [inline README.md:95]:
```suggestion
* `block_all_mixed_content` - this value is deprecated in favor of `upgrade_insecure_requests`. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content for more information.
```

Over aggressive Cmd+F here, the actual URL here uses capitalization.

File: README.md [hunk 3]
```diff
@@ -92,19 +92,19 @@ end
...
 ## Default values
 
 All headers except for PublicKeyPins and ClearSiteData have a default value. The default set of headers is:
 
 ```
-Content-Security-Policy: default-src 'self' https:; font-src 'self' https: data:; img-src 'self' https: data:; object-src 'none'; script-src https:; style-src 'self' https: 'unsafe-inline'
-Strict-Transport-Security: max-age=631138519
+content-security-policy: default-src 'self' https:; font-src 'self' https: data:; img-src 'self' https: data:; object-src 'none'; script-src https:; style-src 'self' https: 'unsafe-inline'
+strict-transport-security: max-age=631138519
 X-Content-Type-Options: nosniff
```
Comment (rzhade3) [inline README.md:104]:
```suggestion
x-content-type-options: nosniff
```

File: README.md [hunk 4]
```diff
@@ -92,19 +92,19 @@ end
...
-X-Download-Options: noopen
+x-download-options: noopen
 X-Frame-Options: sameorigin
```
Comment (rzhade3) [inline README.md:106]:
```suggestion
x-frame-options: sameorigin
```

File: README.md [hunk 5]
```diff
@@ -92,19 +92,19 @@ end
...
-X-Permitted-Cross-Domain-Policies: none
+x-permitted-cross-domain-policies: none
 X-Xss-Protection: 0
```
Comment (rzhade3) [inline README.md:108]:
```suggestion
x-xss-protection: 0
```

File: lib/secure_headers/railtie.rb [hunk 6]
```diff
@@ -5,10 +5,10 @@ module SecureHeaders
     class Railtie < Rails::Railtie
       isolate_namespace SecureHeaders if defined? isolate_namespace # rails 3.0
       conflicting_headers = ["X-Frame-Options", "X-XSS-Protection",
```
Comment (rzhade3) [inline lib/secure_headers/railtie.rb:7]:
```suggestion
      conflicting_headers = ["x-frame-options", "x-xss-protection",
```

File: lib/secure_headers/railtie.rb [hunk 7]
```diff
@@ -5,10 +5,10 @@ module SecureHeaders
...
-                             "X-Permitted-Cross-Domain-Policies", "X-Download-Options",
-                             "X-Content-Type-Options", "Strict-Transport-Security",
-                             "Content-Security-Policy", "Content-Security-Policy-Report-Only",
-                             "Public-Key-Pins", "Public-Key-Pins-Report-Only", "Referrer-Policy"]
+                             "x-permitted-cross-domain-policies", "x-download-options",
+                             "X-Content-Type-Options", "strict-transport-security",
```
Comment (rzhade3) [inline lib/secure_headers/railtie.rb:9]:
```suggestion
                             "x-content-type-options", "strict-transport-security",
```

File: lib/secure_headers/railtie.rb [hunk 8]
```diff
@@ -5,10 +5,10 @@ module SecureHeaders
...
+                             "content-security-policy", "content-security-policy-report-only",
+                             "Public-Key-Pins", "Public-Key-Pins-Report-Only", "referrer-policy"]
```
Comment (rzhade3) [inline lib/secure_headers/railtie.rb:11]:
```suggestion
                             "public-key-pins", "public-key-pins-report-only", "referrer-policy"]
```

**Review by rzhade3 — CHANGES_REQUESTED** [review]
Thanks for this PR! This is a great change. Couple changes suggested below after which we can merge this PR in:

File: CHANGELOG.md (hunk 1 above)
Comment (rzhade3) [reply CHANGELOG.md]: Addressed in https://github.com/github/secure_headers/pull/533/commits/8c5119668de47e00576908c879fa66a8fb41b9fa

**Comment by arashnd** [comment]
thanks! @rzhade3

**Review by KyFaSt — APPROVED** [review]

**Pull request merged**